### ISSUE TITLE
Regression: `ArrayList` objects defined in XAML are added to an `ItemsControl` as one single item

### ISSUE THREAD

**Issue by tomenscape** [AvaloniaUI/Avalonia#10937]
**Describe the bug**
This bug can be seen in the "ViewBox" page of the control gallery. The two combo boxes have just a single "System.Collections.ArrayList" item.

Removing the definition of the `ArrayList` from the XAML document and leaving just the items within it causes those items to appear individually in the combo boxes. This resolves the issue, but was not necessary before.

**Expected behavior**
Any collection provided in this context should be unwrapped and its items individually added to the `ItemsControl`.

**Screenshots**
![image](https://user-images.githubusercontent.com/109803929/230464917-3d5284b3-0e5e-46e7-92ff-a95dba33a785.png)

**Desktop (please complete the following information):**

- OS: Windows
- Version git master

**Comment by grokys** [comment]
I don't think this is a bug. For example in WPF you can see the same behavior:

```
<ItemsControl>
    <col:ArrayList>
      <sys:String>Hello</sys:String>
      <sys:String>World</sys:String>
    </col:ArrayList>
  </ItemsControl>
  ```

![image](https://user-images.githubusercontent.com/1775141/230626052-012bbd5c-fa2e-4afd-848b-8e4caff0ffd2.png)

What happened is:

- Previously `Items` could be reassigned to any `IList`, so the XAML compiler was treating the XAML as meaning "Assign this `ArrayList` to `Items`
- Now `Items` is read-only, so the `ArrayList` can't be assigned. The XAML syntax above has two meanings depending on whether the property is read/write or read-only. In the case of a read-only property the above syntax means "Add the following items to `Items`

tldr; XAML overloads this syntax based on the writability of the target property. Making the property read-only switches from one syntax to another.

The correct fix is to remove the `ArrayList` or wrap it in `<ComboBox.ItemsSource>`.

**Comment by grokys** [comment]
Just to add: I closed this issue because it's not a bug, rather a consequence of the breaking change that changes `Items` to be read-only. Whether we _do_ want to make that breaking for 11.0 is a different question ;)

**Comment by tomenscape** [comment]
Alright, but the control gallery still needs to be fixed.
